core/controllers/feedback_test.py

```diff
@@ -336,3 +344,243 @@ def _get_email(n):
+            '%s/%s/%s' % (feconf.FEEDBACK_THREAD_URL_PREFIX,
+                          self.EXP_ID, thread_id_1.split('.')[1]))
+        self.assertEqual(
+            response_dict['suggestion']['id'], thread_id_1)
+
+        # Test invalid list type in GET request. We are using testapp.get()
+        # here because get_json() with expect_errors=True does not work as
+        # expected. Please look at the doc string for get_json() for details.
+        response_dict = self.testapp.get(
+            '%s/%s?list_type=%s&has_suggestion=%s' % ( 
+            feconf.SUGGESTION_LIST_URL_PREFIX, self.EXP_ID, 'clsed', 'true'),
+            status=400, expect_errors=True)
+        self.assertEqual(response_dict.status_int, 400)
+
+    def _accept_suggestion(self, thread_id, method_to_call, csrf_token,
+                           expect_errors=False, expected_status_int=200):
+        with self.swap(exp_domain.Exploration, '_verify_all_states_reachable',
+                       self._return_null):
+            with self.swap(exp_domain.Exploration, '_verify_no_dead_ends',
+                           self._return_null):
+                return self.put_json(
+                    '%s/%s/%s' % (feconf.SUGGESTION_ACTION_URL_PREFIX,
+                                  self.EXP_ID, thread_id.split('.')[1]), 
+                    {'action': u'accept'}, csrf_token,
+                    expect_errors=expect_errors, 
+                    expected_status_int=expected_status_int)
+
+    def _reject_suggestion(self, thread_id, method_to_call, csrf_token,
+                           expect_errors=False, expected_status_int=200):
+        with self.swap(rights_manager.Actor, 'can_edit',
+                       method_to_call):
+            return self.put_json(
+                '%s/%s/%s' % (feconf.SUGGESTION_ACTION_URL_PREFIX,
+                              self.EXP_ID, thread_id.split('.')[1]), 
+                {'action': u'reject'}, csrf_token, expect_errors=expect_errors,
+                expected_status_int=expected_status_int)
+    
+    def test_action_suggestions(self):
+        self._create_suggestions()
+        self.login(self.EDITOR_EMAIL)
+        csrf_token = self.get_csrf_token_from_response(
+            self.testapp.get('/create/%s' % self.EXP_ID))
+        response_dict = self.get_json(
+            '%s/%s?list_type=%s&has_suggestion=%s' % ( 
+            feconf.SUGGESTION_LIST_URL_PREFIX, self.EXP_ID, 'all', 'true'))
+        threads = response_dict['threads']
+        thread_id_1 = threads[0]['thread_id']
+        thread_id_2 = threads[1]['thread_id']
+        thread_id_3 = threads[2]['thread_id']
+
+        # Accept a suggestion.
+        self._accept_suggestion(thread_id_1, self._return_true, csrf_token)
+        updated_exploration = exp_services.get_exploration_by_id(self.EXP_ID)
+        self.assertEqual(
+            updated_exploration.states['State A'].content[0].to_dict(),
+            feedback_services.get_suggestion(
```

Comment:
Hm. I think it's better for this to be just a constant. Maybe in the setUp() you could have the text of the suggestion be 'new accepted suggestion for state A' or something like that.